template/template.go

```diff
@@ -20,12 +20,12 @@ import (
 	"net/url"
 	"path"
 	"path/filepath"
-	"regexp"
 	"sort"
 	"strings"
 	tmpltext "text/template"
 	"time"
 
+	"github.com/grafana/regexp"
```

Comment:
:thinking:

Reply:
This is in line with https://github.com/prometheus/prometheus/pull/10251

Reply:
I believe this is only temporary, has Bryan has submitted all of the patches of grafana/regexp to go already.

Reply:
I don't know if that is needed for alertmanager.

Reply:
Right, I don't think we've dealt with performance issues in the area so I'm indifferent about the change.

The only thing to consider here would be @metalmatze's point on the alignment with Prometheus which I guess is a net positive?

Reply:
That's a good shout @roidelapluie. @metalmatze can we remove this dependency and stick with Go's regexp then?

Reply:
Yes, this is only temporary indeed. I'll remove it then and stick with stdlib. No problem. 
AFAIK the package is mostly for Loki dealing with lots of regexp matches in each log line.